library/cartridge_connect_to_membership.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+
+import pkgutil
+
+if pkgutil.find_loader('ansible.module_utils.helpers'):
+    import ansible.module_utils.helpers as helpers
+else:
+    import module_utils.helpers as helpers
+
+argument_spec = {
+    'console_sock': {'required': True, 'type': 'str'},
+    'hostvars': {'required': True, 'type': 'dict'},
+    'play_hosts': {'required': True, 'type': 'list'},
+}
+
+
+def probe_server(control_console, advertise_uris):
+    return control_console.eval_res_err('''
+        local advertise_uris = ...
+
+        local cartridge = require('cartridge')
+        local fiber = require('fiber')
+
+        local count = 0
+        local channel = fiber.channel()
+
+        for _, advertise_uri in ipairs(advertise_uris) do
+            count = count + 1
+
+            fiber.create(function()
+                local _, err = cartridge.admin_probe_server(advertise_uri)
+                channel:put({advertise_uri, err})
+            end)
+        end
+
+        local errors = {}
+        for _ = 1, count do
+            local res = channel:get(60)
+            if res == nil then
+                table.insert(errors, {nil, 'Impossible to get probe result: timeout'})
+                break
+            end
+            if res[2] ~= nil then
+                local err = res[2].err or res[2]
+                table.insert(errors, {res[1], ("Impossible to probe server '%%s': %s"):format(err)})
+            end
+        end
+
+        if next(errors) ~= nil then
+            return nil, errors
+        end
+
+        return true
+    ''', advertise_uris)
+
+
+def connect_to_membership(params):
+    control_console = helpers.get_control_console(params['console_sock'])
+    hostvars = params['hostvars']
+    play_hosts = params['play_hosts']
+
+    advertise_uris = []
+    instance_names = {}
+
+    for instance_name, instance_vars in hostvars.items():
+        if helpers.is_expelled(instance_vars) or helpers.is_stateboard(instance_vars):
+            continue
+
+        if 'config' not in instance_vars or 'advertise_uri' not in instance_vars['config']:
+            continue
+
+        advertise_uri = instance_vars['config']['advertise_uri']
+        advertise_uris.append(advertise_uri)
+        instance_names[advertise_uri] = instance_name
+
+    res, errors = probe_server(control_console, advertise_uris)
```

Comment:
Let's rename this function.

What about `probe_instances` or `connect_instances`?